File: 02_reverse_string.cpp
#include <bits/stdc++.h>
using namespace std;

/*
Task 2: Reverse a string.
Reads a line and prints its reverse.
*/

int main()
{
    string s;
    if (!getline(cin, s))
        return 0;
    for (int i = (int)s.size() - 1; i >= 0; --i) cout << s[i];
    cout << '\n';
    return 0;
}

// Time Complexity: O(n)
// Space Complexity: O(1)

File: 02_reverse_string.py
"""
Task 2: Reverse a string.
Reads a line and prints its reverse.
"""
import sys

def reverse_string(s: str) -> str:
    return ''.join(s[i] for i in range(len(s)-1, -1, -1))

if __name__ == '__main__':
    s = sys.stdin.readline().rstrip('\n')
    print(reverse_string(s))

# Time Complexity: O(n)
# Space Complexity: O(n)  # for the reversed string
